Mathlib/FieldTheory/PrimitiveElement.lean

```diff
@@ -355,3 +355,68 @@ theorem AlgHom.card (F E K : Type*) [Field F] [Field E] [Field K] [IsAlgClosed K
   convert (AlgHom.card_of_powerBasis (L := K) (Field.powerBasisOfFiniteOfSeparable F E)
     (IsSeparable.separable _ _) (IsAlgClosed.splits_codomain _)).trans (PowerBasis.finrank _).symm
 #align alg_hom.card AlgHom.card
+
+section iff
+
+namespace Field
+
+open FiniteDimensional IntermediateField Polynomial Algebra Set
+
+variable (F : Type*) {E : Type*} [Field F] [Field E] [Algebra F E] [FiniteDimensional F E]
+
+theorem primitive_element_of_minpoly_natDegree_eq (α : E) :
+    F⟮α⟯ = ⊤ ↔ (minpoly F α).natDegree = finrank F E := by
+  rw [← adjoin.finrank (IsIntegral.of_finite F α), ← finrank_top F E]
+  refine ⟨fun h => ?_, fun h => ?_⟩
+  · exact congr_arg (fun K : IntermediateField F E => finrank F K) h
+  · exact eq_of_le_of_finrank_eq le_top h
+
+theorem primitive_element_of_minpoly_degree_eq (α : E) :
+    F⟮α⟯ = ⊤ ↔ (minpoly F α).degree = finrank F E := by
+  rw [degree_eq_iff_natDegree_eq, primitive_element_of_minpoly_natDegree_eq]
+  exact minpoly.ne_zero_of_finite F α
+
+variable [IsSeparable F E] (A : Type*) [Field A] [IsAlgClosed A] [Algebra F A]
```

Comment:
Thanks for the review. I have pushed a new version that requires only normal closure. I had to add a couple of results about `IsNormalClosure`. I am not sure about `IsNormalClosure.algHom_equiv_algHom_isAlgClosed` since the equiv is completely non explicit so maybe I should just incorporate that in the proof of `IsNormalClosure.algHom.card` where it is used

Reply:
Not sure `IsNormalClosure` is suitable here (it won't apply to the algebraic closure, for example). The right condition is that every minimal polynomial of `E` over `F` splits in `A`, which appears a lot in the NormalClosure file, and @acmepjz has suggested making a def `IsSplitIn` for that.

Reply:
Yeah, I think you are right. I'll refactor the code again.

Reply:
Oops, that is indeed much better. Sorry for making an unhelpful suggestion first time round.